src/main/java/org/radarcns/management/security/ClaimsTokenEnhancer.java

```diff
@@ -59,7 +60,9 @@ public OAuth2AccessToken enhance(OAuth2AccessToken accessToken,
             Optional<User> optUser = userRepository.findOneByLogin(userName);
             if (optUser.isPresent()) {
                 List<String> roles = optUser.get().getRoles().stream()
-                        .filter(role -> Objects.nonNull(role.getProject()))
+                        .filter(role -> Objects.nonNull(role.getProject())
+                                && !AuthoritiesConstants.INACTIVE_PARTICIPANT.equals(role
```

Comment:
Shouldn't we include the INACTIVE_PARTICIPANT role in the JWT anyway? If the token does not contain the role, then the user will not be able to get access to old project info based on this role.

Reply:
I don't think so. I would consider a token to have information that could only allow them to get certain access, not the data from history that is not valid anymore.